### ISSUE TITLE
Deserialization of enums doesn't work

### ISSUE THREAD

**Issue by vorner** [dtolnay/erased-serde#8]
It seems the deserializer gets confused about types of enums. This code demonstrates the problem. Normal deserialization (in the `normal` test) works well, and deserializing some special types (like `IpAddr`) also works. But when I try to deserialize an enum from my code, it fails with the error `Error { msg: "invalid type: enum, expected enum X" }`

```rust
  extern crate serde;
  #[macro_use]
  extern crate serde_derive;
  extern crate serde_json;
  extern crate erased_serde;
  
  #[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
  enum X {
      A, 
      B, 
  }   
  
  #[cfg(test)]
  mod tests {
      use std::net::IpAddr;
  
      use serde::Deserialize;
                   
      use super::*;
                   
      #[test]      
      fn normal() {
          let mut deserializer = serde_json::Deserializer::from_str("\"A\"");
          assert_eq!(X::A, X::deserialize(&mut deserializer).unwrap());
      }            
                   
      #[test]      
      fn erased() {
          let mut deserializer = serde_json::Deserializer::from_str("\"A\"");
          let mut erased = erased_serde::Deserializer::erase(&mut deserializer);
          assert_eq!(X::A, erased_serde::deserialize::<X>(&mut erased).unwrap());
      }            
              
      #[test] 
      fn works() {
          let mut deserializer = serde_json::Deserializer::from_str("\"192.0.2.1\"");
          let mut erased = erased_serde::Deserializer::erase(&mut deserializer);
          let expected = "192.0.2.1".parse::<IpAddr>().unwrap();
          assert_eq!(expected, erased_serde::deserialize::<IpAddr>(&mut erased).unwrap());
      }                                               
  }   
```

**Comment by dtolnay** [comment]
Thanks for reporting this! I released a fix in 0.3.1 and added better testing.

**Comment by vorner** [comment]
It always amazes me how fast some people respond :-). Thank you
